processor/spanmetricsprocessor/config.go

```diff
@@ -45,4 +45,16 @@ type Config struct {
 	// The dimensions will be fetched from the span's attributes. Examples of some conventionally used attributes:
 	// https://github.com/open-telemetry/opentelemetry-collector/blob/main/model/semconv/opentelemetry.go.
 	Dimensions []Dimension `mapstructure:"dimensions"`
+
+	// MetricKeyToDimensionsCacheSize defines the size of the LRU cache for storing MetricKeyToDimensions cache
+	MetricKeyToDimensionsCacheSize int `mapstructure:"metric_key_to_dimensions_cache_size"`
+}
+
+// GetMetricKeyToDimensionsCacheSize get the value from config for the length of LRU cache `MetricKeyToDimensionsCacheSize`
+// if the valve from the config is invalid, return the default value.
+func (c Config) GetMetricKeyToDimensionsCacheSize() int {
+	if c.MetricKeyToDimensionsCacheSize <= 0 {
```

Comment:
0 should be a valid size imo, i.e no cache

Reply:
and should the last return be returning the config size?

Reply:
When we create an LRU cache we have to specify a size, it's also the behaviour of the LRU library:

```
// github.com/hashicorp/golang-lru/simplelru/lru.go
	if size <= 0 {
		return nil, errors.New("Must provide a positive size")
	}
```

Reply:
> 

fixed

Reply:
> and should the last return be returning the config size?

fixed

Reply:
>  When we create an LRU cache we have to specify a size, it's also the behaviour of the LRU library

interesting, would be good to add a bypass that skips the caching mechanism to make things wholistic but not major. We can add it in later since it's not impacting current functionality which doesn't allow you to disable the cache. This looks fine to me for now 👍

Reply:
I agree, there should be no cache if it is unspecified, this preserves current behaviour and removes risk of breaking change.

Reply:
> it's not impacting current functionality

Maybe you're right though, it will reset cumulatives, but no end-user breaking change.

Reply:
And I also suppose this was planned to be addressed by the linked issue.